### PR TITLE
[TA] Analyze Healthcare entities updates

### PR DESCRIPTION
**Pull request by mssfang**
Include all updated changes to healthcare analysis endpoint.

fixes: https://github.com/Azure/azure-sdk-for-java/issues/18722

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ReadmeSamples.java [hunk 1]
```diff
@@ -191,35 +195,43 @@ public void recognizeHealthcareEntities() {
...
-                                healthcareEntityLink.getDataSourceId(), healthcareEntityLink.getDataSource()));
+                healthcareEntitiesResult.getEntities().forEach(healthcareEntity -> {
+                    System.out.printf("\ti = %d, Text: %s, category: %s, subcategory: %s, confidence score: %f.%n",
+                        ct.getAndIncrement(), healthcareEntity.getText(), healthcareEntity.getCategory(),
+                        healthcareEntity.getSubcategory(), healthcareEntity.getConfidenceScore());
+                    IterableStream<EntityDataSource> healthcareEntityDataSources =
+                        healthcareEntity.getDataSources();
+                    if (healthcareEntityDataSources != null) {
+                        healthcareEntityDataSources.forEach(healthcareEntityLink -> System.out.printf(
+                            "\t\tHealthcare data source ID: %s, data source: %s.%n",
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ReadmeSamples.java:226]: consider changing to `Entity ID in data source`

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientJavaDocCodeSnippets.java [hunk 2]
```diff
@@ -808,57 +813,63 @@ public void analyzeHealthcareMaxOverload() {
...
-        RecognizeHealthcareEntityOptions options = new RecognizeHealthcareEntityOptions()
+        AnalyzeHealthcareEntitiesOptions options = new AnalyzeHealthcareEntitiesOptions()
             .setIncludeStatistics(true);
 
-        textAnalyticsAsyncClient.beginAnalyzeHealthcare(documents, options)
-            .flatMap(AsyncPollResponse::getFinalResult)
+        textAnalyticsAsyncClient.beginAnalyzeHealthcareEntities(documents, options)
+            .flatMap(pollResult -> {
+                AnalyzeHealthcareEntitiesOperationDetail operationResult = pollResult.getValue();
+                System.out.printf("Job created time: %s, expiration time: %s.%n",
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientJavaDocCodeSnippets.java:822]: operation time . same in other places

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ReadmeSamples.java (hunk 1 above)
Comment (maririos) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ReadmeSamples.java:226]: same in other places

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/lro/CancelAnalyzeHealthcareEntities.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    + " his anginal equivalent. Due to the patient's increased symptoms and family history and"
+                    + " history left main disease with total occasional of his RCA was referred"
+                    + " for revascularization with open heart surgery."));
+        }
+
+        SyncPoller<AnalyzeHealthcareEntitiesOperationDetail, PagedIterable<AnalyzeHealthcareEntitiesResultCollection>>
+            syncPoller = client.beginAnalyzeHealthcareEntities(documents, null, Context.NONE);
+
+        PollResponse<AnalyzeHealthcareEntitiesOperationDetail> pollResponse = syncPoller.poll();
+        System.out.printf("The Job ID that is cancelling is %s.%n", pollResponse.getValue().getOperationId());
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/lro/CancelAnalyzeHealthcareEntities.java:60]: operation id?

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeHealthcareEntityAsyncClient.java [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+            requestStatistics.getErroneousDocumentsCount(), requestStatistics.getTransactionsCount()
+        );
+        AnalyzeHealthcareEntitiesResultCollectionPropertiesHelper.setStatistics(
+            analyzeHealthcareEntitiesResultCollection, batchStatistic);
+
+        final List<TextAnalyticsError> errors = healthcareJobState.getErrors();
+
+        if (!CoreUtils.isNullOrEmpty(errors)) {
+            final TextAnalyticsException textAnalyticsException = new TextAnalyticsException(
+                "Analyze operation failed", null, null);
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeHealthcareEntityAsyncClient.java:195]: I can update it to it.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/AnalyzeHealthcareEntitiesOperationDetailPropertiesHelper.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    /**
+     * Type defining the methods to set the non-public properties of an
+     * {@link AnalyzeHealthcareEntitiesOperationDetail}
+     * instance.
+     */
+    public interface AnalyzeHealthcareEntitiesOperationDetailAccessor {
+        void setOperationId(AnalyzeHealthcareEntitiesOperationDetail operationDetail, String operationId);
+        void setCreatedAt(AnalyzeHealthcareEntitiesOperationDetail operationDetail, OffsetDateTime createdAt);
+        void setExpiresAt(AnalyzeHealthcareEntitiesOperationDetail operationDetail, OffsetDateTime expiresAt);
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/AnalyzeHealthcareEntitiesOperationDetailPropertiesHelper.java:23]: If service return null value, the expiresAt will be null. FYI. this setter method is used internally for not introduce the constructor of ` AnalyzeHealthcareEntitiesOperationDetail`.

**Review by maririos — APPROVED** [review]
Design LGTM

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/HealthcareEntityPropertiesHelper.java [hunk 6]
```diff
@@ -44,7 +46,7 @@ public static void setText(HealthcareEntity healthcareEntity, String text) {
         accessor.setText(healthcareEntity, text);
     }
 
-    public static void setCategory(HealthcareEntity healthcareEntity, EntityCategory category) {
+    public static void setCategory(HealthcareEntity healthcareEntity, String category) {
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/HealthcareEntityPropertiesHelper.java:49]: why did we drop the enum?

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/lro/AnalyzeHealthcareEntitiesAsync.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+                        });
+                    });
+                }
+            ));
+
+        // The .subscribe() creation and assignment is not a blocking call. For the purpose of this example, we sleep
+        // the thread so the program does not end before the send operation is complete. Using .block() instead of
+        // .subscribe() will turn this into a synchronous call.
+        try {
+            TimeUnit.MINUTES.sleep(20);
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/lro/AnalyzeHealthcareEntitiesAsync.java:106]: nit: 20 mins seems too much, 5 minutes should be enough?

**Review by samvaity — APPROVED** [review]

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/HealthcareEntityPropertiesHelper.java (hunk 6 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/HealthcareEntityPropertiesHelper.java:49]: Healthcare entity category is not same as the Named(Categorized) Entity category. Agreed with .NET approach.

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 8]
```diff
@@ -315,42 +315,50 @@ Text Analytics for health is a containerized service that extracts and labels re
...
+        + "The patient is a 54-year-old gentleman with a history of progressive angina over the past"
+        + " several months. The patient had a cardiac catheterization in July of this year revealing total"
+        + " occlusion of the RCA and 50% left main disease , with a strong family history of coronary"
+        + " artery disease with a brother dying at the age of 52 from a myocardial infarction and another"
+        + " brother who is status post coronary artery bypass grafting. The patient had a stress"
+        + " echocardiogram done on July , 2001 , which showed no wall motion abnormalities,"
+        + " but this was a difficult study due to body habitus. The patient went for six minutes with"
+        + " minimal ST depressions in the anterior lateral leads , thought due to fatigue and wrist pain,"
+        + " his anginal equivalent. Due to the patient's increased symptoms and family history and"
+        + " history left main disease with total occasional of his RCA was referred"
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/README.md:334]: Service team suggested to use it.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeHealthcareEntityAsyncClient.java [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+        return Mono.just(new PollResponse<>(status, operationResultPollResponse.getValue()));
+    }
+
+    private Context getNotNullContext(Context context) {
+        return context == null ? Context.NONE : context;
+    }
+
+    private AnalyzeHealthcareEntitiesOptions getNotNullAnalyzeHealthcareEntitiesOptions(
+        AnalyzeHealthcareEntitiesOptions options) {
+        return options == null ? new AnalyzeHealthcareEntitiesOptions() : options;
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeHealthcareEntityAsyncClient.java:329]: Created an issue to address other options bags. #19018

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java [hunk 10]
```diff
@@ -1019,37 +1019,77 @@ public String getDefaultLanguage() {
...
+            }), options);
+    }
+
+    /**
+     * Analyze healthcare entities, entity data sources, and entity relations in a list of
+     * {@link TextDocumentInput documents} with provided request options.
+     *
+     * Note: In order to use this functionality, request to access public preview is required.
+     * Azure Active Directory (AAD) is not currently supported. For more information see
+     * <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/how-tos/text-analytics-for-health?tabs=ner#request-access-to-the-public-preview">this</a>.
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java:1065]: Already mentioned in the README Healthcare section

**Pull request merged**